File: Kylie_Guzman/henry_guzman/mis_programas/dart/piedra_papel_tijera.dart
import 'dart:math';
import 'dart:io';

void main(List<String> arguments) {

  var users_wins = 0;
  var computer_wins = 0;
  var empates = 0;

  List opciones = ['piedra', 'papel', 'tijera'];

  while(true){

    print ('elige piedra, papel, tijera o presione la tecla "x" para salir del juego.');
    var eleccion_users = stdin.readLineSync()!.toLowerCase();

    if (eleccion_users.toString() == 'x' ){
      break; 

    }if (!opciones.contains(eleccion_users)){
      print ('ERROR!');
      continue;
    }

    Random random = Random();
    var random_num = random.nextInt(3);
    var eleccion_ordenador = opciones[random_num];

    print('el ordenador eligio $eleccion_ordenador');

    if (eleccion_users =='piedra' && eleccion_ordenador == 'tijera'){

      print ('bien hecho!');
      users_wins ++;

    }else if (eleccion_users =='papel' && eleccion_ordenador == 'piedra'){

      print ('bien hecho!');
      users_wins ++;

    }else if (eleccion_users =='tijera' && eleccion_ordenador == 'papel'){

      print ('bien hecho!');
      users_wins ++;

    }else if (eleccion_users == eleccion_ordenador){
    
      print ('ha sido un empate');
      empates ++;
    
    }else if (eleccion_users =='tijera' && eleccion_ordenador == 'piedra'){
      
      print('puedes hacerlo mejor');
      computer_wins ++;

    }else if (eleccion_users =='piedra' && eleccion_ordenador == 'papel'){
      
      print('puedes hacerlo mejor');
      computer_wins ++;

    }else if (eleccion_users =='papel' && eleccion_ordenador == 'tijera'){
      
      print('puedes hacerlo mejor');
      computer_wins ++;

    };

  }
  print ('tu = $users_wins puntos');
  print ('el ordenador = $computer_wins puntos');
  print ('empates = $empates');

    if(users_wins > computer_wins){

      print('     has ganado!! ');

    }else if(users_wins < computer_wins){

      print('     has perdido :( ');

    }else (){

      print ('ha sido un empate');

    };
}

File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/python/practicas2/piedra_papel_tijera.py
import random

piedra = 1
papel = 2
tijera = 3

userwins = 0
computerwins = 0
empates = 0
total = 0

while (True):

    print( "Elige piedra papel o tijera:")
    print( "1- piedra")
    print( "2- papel")
    print( "3- tijera")
    print( "7- para salir del juego")
    jugador = int(input())


    if jugador == 7:
        break

    if jugador == 1:
        print( "jugador escoge: Piedra")
    if jugador == 2:
        print( "jugador escoge: papel")
    if jugador == 3:
        print( "jugador escoge: tijera")

    computador = random.randrange(1,3)

    if computador == 1:
        print( "computador escoge: Piedra")
    if computador == 2:
        print( "computador escoge: papel")
    if computador == 3:
        print( "computador escoge: tijera")


    if jugador == 1 and computador == 3 or jugador == 2 and computador == 1 or jugador == 3 and computador == 2:
        print( "Ganaste")
        userwins = userwins + 1
        total = total + 1
    if jugador == 1 and computador == 1 or jugador == 2 and computador == 2 or jugador == 3 and computador == 3:
        print( "Empate")
        empates = empates + 1
        total = total + 1
    if jugador == 1 and computador == 2 or jugador == 2 and computador == 3 or jugador == 3 and computador == 1:
        print( "Perdiste")
        computerwins = computerwins + 1
        total = total + 1

print( "Victorias del jugador: ",userwins)
print( "Victorias de la computadora: ",computerwins)
print( "Empates: ",empates)
print( "Total de partidas jugadas: ",total)

if userwins > computerwins:
    print( "El ganador es el jugador")
if userwins < computerwins:
    print( "El ganador es la computadora")
if userwins == computerwins:
    print( "La partida ha quedado en empate")